### PR TITLE
first pass on Bug 3964 recovery

### PR DESCRIPTION
**Pull request by maxtaco**
- don't do anything fancy, just try all possible device XORs
- don't try to upgrade the keyring file or the secret store

### CODE REVIEW

**Comment by maxtaco** [comment]
r? @patrickxb 

I think I might just leave this here, and postpone any secret file rewriting until a later job.  So once you're in this situation, you stay there until you change your passhrase or otherwise reencrypt your key.

Note that there is a big problem I just thought of --- this corruption can spread virally across all of your devices, since you can sync your bad secret store to another device, even with a fixed client, via the same mechanism.  I'm not quite sure how to solve that situation in general. But weirdly, this recover codepath will solve that problem (unintentionally).

**Comment by maxtaco** [comment]
C/I failures are unrelated KBFS failures :/

**Comment by patrickxb** [comment]
So if someone is stuck, we tell them to change passphrase?

**Comment by maxtaco** [comment]
No, with this patch, it "just works."   We try to decrypt with what we guess is the corrupted fullSecret.  I did actually test it successfully on one of my repro cases.

What this patch doesn't try to do is to fix the situation going forward.  It always applies the recovery option.

File: go/libkb/lksec.go [hunk 1]
```diff
@@ -326,25 +326,55 @@ func (s *LKSec) Encrypt(src []byte) (res []byte, err error) {
...
+			s.G().Log.Debug("| Success")
+			return res, s.ppGen, serverHalf, nil
+		}
+	}
+
+	err = PassphraseError{"failed to open secretbox"}
+	return nil, 0, LKSecServerHalf{}, err
+}
+
+func (s *LKSec) Decrypt(lctx LoginContext, src []byte) (res []byte, gen PassphraseGeneration, erroneousMask LKSecServerHalf, err error) {
```
Comment (maxtaco) [inline go/libkb/lksec.go]: i was thinking in follow-on PRs it would be.

**Pull request merged**